TestLogic: improper/incomplete test for timezone errors

The `datetime` test in TestLogic expects an incorrect/invalid timezone to fail with "cannot find ..." from Go's time library. However due to the way Go's time library is written, combined with how the distrib's standard package installs Go on (at least) Ubuntu 15, this error cannot be produced upon selecting an invalid timezone; instead the error shows up as "unable to find zoneinfo.zip", and thus:

```
testdata/datetime:309: SET TIME ZONE 'foobar'
--- FAIL: TestLogic (1.89s)
        logic_test.go:459: testdata/datetime:309: expected "cannot find foobar.*", but found
                pq: open /usr/lib/go/lib/time/zoneinfo.zip: no such file or directory
```

The problem is that `zoneinfo.zip` is not necessary on Unix (and thus is not installed by Go's package on Ubuntu 15), yet time's lookup code will fall back to it if it cannot find the selected timezone in the standard locations. 

The proper fix / workaround is to accept an unavailable zoneinfo.zip as a correct error message for an invalid timezone.

--- a/sql/set.go
+++ b/sql/set.go
@@ -94,7 +94,7 @@ func (p *planner) SetTimeZone(n *parser.SetTimeZone) (planNode, *roachpb.Error) 
 			location = "UTC"
 		}
 		if _, err := time.LoadLocation(location); err != nil {
-			return nil, roachpb.NewError(err)
+			return nil, roachpb.NewUErrorf("cannot find time zone %q: %v", location, err)
 		}
 		p.session.Timezone = &Session_Location{Location: location}
 

--- a/sql/testdata/datetime
+++ b/sql/testdata/datetime
@@ -306,7 +306,7 @@ SELECT '2015-08-24 23:45:45.53453'::timestamp
 ----
 2015-08-25 03:45:45.53453 +0000 +0000
 
-statement error cannot find foobar.*
+statement error cannot find time zone "foobar":.*
 SET TIME ZONE 'foobar'
 
 statement ok

